pyhealth/models/deepr.py

```diff
@@ -1,46 +1,36 @@
-import functools
-from typing import Dict, List, Optional, Tuple
+# Author: Joshua Steier
+# Description: Deepr model implementation for PyHealth 2.0
+
+from typing import Any, Dict, Optional, Tuple
 
 import torch
 import torch.nn as nn
 
-from pyhealth.datasets import BaseEHRDataset
+from pyhealth.datasets import SampleDataset
 from pyhealth.models import BaseModel
+from pyhealth.models.embedding import EmbeddingModel
 
 
 class DeeprLayer(nn.Module):
-    """Deepr layer.
-
-    Paper: P. Nguyen, T. Tran, N. Wickramasinghe and S. Venkatesh,
-    " Deepr : A Convolutional Net for Medical Records," in IEEE Journal
-    of Biomedical and Health Informatics, vol. 21, no. 1, pp. 22-30,
-    Jan. 2017, doi: 10.1109/JBHI.2016.2633963.
-
-    This layer is used in the Deepr model.
+    """Deepr convolution + max pooling over the sequence dimension.
 
     Args:
-        feature_size: embedding dim of codes (m in the original paper).
-        window: sliding window (d in the original paper)
-        hidden_size: number of conv filters (motif size, p, in the original paper)
-    Examples:
-        >>> from pyhealth.models import DeeprLayer
-        >>> input = torch.randn(3, 128, 5)  # [batch size, sequence len, input_size]
-        >>> layer = DeeprLayer(5, window=4, hidden_size=7) # window does not impact the output shape
-        >>> outputs = layer(input)
-        >>> outputs.shape
-        torch.Size([3, 7])
+        feature_size: Input feature size (embedding_dim).
+        window: Sliding window radius d; kernel_size = 2*d + 1.
+        hidden_size: Number of convolution filters.
```

Comment:
Could you update the usage examples in the docstring similar to the old docstring? This could help us to generate a nice document.